superset/sql_parse.py

```diff
@@ -111,7 +111,18 @@ def is_select(self) -> bool:
         return self._parsed[0].get_type() == "SELECT"
 
     def is_explain(self) -> bool:
-        return self.stripped().upper().startswith("EXPLAIN")
+        # Remove comments
+        statements_without_comments = [
+            statement.strip(" \t\n;")
+            for statement in self.stripped().upper().splitlines()
+            if not statement.startswith("--")
+        ]
```

Comment:
This will miss some edge cases like:

```sql
-- This is a comment
 -- this is another comment but with a space in the front
EXPLAIN SELECT * FROM TABLE
```

You need to `lstrip()` each line as well before testing if it `startswith()`:

```python
        statements_without_comments = [
            statement.strip(" \t\n;")
            for statement in self.stripped().upper().splitlines()
            if not statement.lstrip().startswith("--")
        ]
```

Reply:
I find it simpler to just use regular expression:

```python
    queries = re.sub(r'\s*--.*', '', queries)
    queries = [x.strip(' \t\n\r') for x in queries.split(';')]
    queries = [x for x in queries if x]  # query must be non-empty
```